### DISCUSSION TITLE
Working with time series data

### DISCUSSION THREAD

**Discussion by koresaram777** [feature-engine/feature_engine#278] [Q&A] [answered]
Hello! Thank you very much for such great library, which helps in my daily work. Also, I have one question. Can I use feature selection techniques and missing data imputation techniques, when I deal with time series data? Because in internet, I have seen that for time series data, people use different techniques.

**Comment by solegalli** [comment, marked as answer]
Hey @koresaram777 

**Regarding selection methods:**
I’ve not come across feature selection methods tailored for time series. Once you’ve transformed the data set into a regression or classification problem you should be able to use standard feature selection methods, including those available in Featire-engine. 

Might need to be mindful that some features will be highly correlated (e.g., sales yesterday, average sales over past week etc.). 

Answer powered by @KishManani 

**Regarding feature engineering methods:**
After you transformed the data into a regression or classification problem, creating features like average sales in the past 3 months, or maximum payment in the year, or time since last transaction. Then you can use the feature engineering techniques (including missing data filling) available in Feature engine.

Now to fill missing data in the time series itself, there are other methods, like forward or backwards fill, linear or spline interpolation, that are not available at the moment in Feature-engine. The same to remove outliers. For a time series, you would normally look for outliers in a window, not in the entire data, so that is at the moment not supported by feature engine.

But I would like to make it part of the library. It is just that I would need support from the community, or be able to organise some sort of hackathon, because these are some big additions.

Feel free to make specific suggestions either here or through the issues for specific requests.

Hope this answers the question. 

Best wishes

**Reply by koresaram777** [reply]
Thank you very much for the answer! I will be waiting for the update, in my opinion, it would be very useful, if users can preprocess time series data using feature-engine. In my case, I am working in industrial area (smart factory, smart city and related stuff), and almost all data is time series.
